src/components/reusables/ActionBar/ActionBar.tsx

```diff
@@ -1,63 +1,116 @@
-  title: string;
-  subtitle?: string;
-  icon: ActionIconVariant;
-  actionCallBack: () => void;
-}
-
-export enum ActionIconVariant {
-  Add,
-  Remove,
+  actions?: React.ReactNode;
+  additionalText?: React.ReactNode;
+  children?: React.ReactNode;
+  color?: 'light' | 'neutral' | 'warning' | 'success' | 'danger';
+  onClick?: ClickHandler;
+  open?: boolean;
+  subtitle?: React.ReactNode;
+  title?: React.ReactNode;
 }
 
 export const ActionBar = ({
-  title,
+  actions,
+  additionalText,
+  children,
+  color = 'neutral',
+  onClick,
+  open = false,
   subtitle,
-  icon = ActionIconVariant.Add,
-  actionCallBack,
+  title,
 }: ActionBarProps) => {
-  let button;
-  switch (icon) {
-    case ActionIconVariant.Add:
-      button = (
-        <Button
-          className={classes.actionButton}
-          icon={<AddCircle />}
-          variant={ButtonVariant.Quiet}
-          color={ButtonColor.Success}
-          onClick={actionCallBack}
-          aria-label={'add'}
-        ></Button>
-      );
-      break;
-    case ActionIconVariant.Remove:
-      button = (
-        <Button
-          className={classes.actionButton}
-          icon={<MinusCircle />}
-          variant={ButtonVariant.Quiet}
-          color={ButtonColor.Danger}
-          onClick={actionCallBack}
-          aria-label={'remove'}
-        ></Button>
-      );
-      break;
-  }
+  const headerId = useId();
+  const contentId = useId();
+
   return (
-    <div className={classes.actionBar}>
-      <div>
-        <h4 className={classes.title}>{title}</h4>
-        {subtitle && <div className={classes.subtitle}>{subtitle}</div>}
-      </div>
-      <div className={classes.actionButtonContainer}>{button}</div>
-    </div>
+    <>
+      <ActionBarContext.Provider
+        value={{
+          onClick,
+          open,
+          headerId,
+          contentId,
+          color,
+        }}
+      >
+        <div
+          className={cn(classes.actionBar, classes[color], {
+            [classes.subtitle]: subtitle,
+            [classes.open]: open,
+            [classes.clickable]: onClick,
+          })}
+          tabIndex={-1}
+          onClick={onClick}
+        >
+          {children && (
+            <div className={cn(classes.actionBarIcon)}>
+              <ActionBarIcon />
+            </div>
+          )}
+          {children ? (
+            <button
+              className={cn(classes.actionBarHeader, classes.clickable)}
+              type='button'
+              onClick={onClick}
+              id={headerId}
+              data-testid='action-bar'
+              aria-expanded={open}
+              aria-controls={contentId}
+            >
+              <div className={classes.actionBarTexts}>
+                <div className={classes.title}>{title}</div>
+                <div className={classes.subtitle}>{subtitle}</div>
+              </div>
+            </button>
+          ) : (
+            <button
+              className={cn(classes.actionBarHeader)}
```

Comment:
This one does not have the clickable class. Does that mean it also doesn't get the cursor: pointer from actionHeader.clickable ?